### PR TITLE
Adds Redis session support behind env var

### PR DESCRIPTION
**Pull request by gidsg**
https://trello.com/c/XXL04KtG/379-3-run-the-new-user-session-management-on-the-local-development-environment
This adds Redis session support behind an env var so that we can manage the rollout.
Code based largely on @lfdebrux's prior spike.

### CODE REVIEW

File: dmutils/flask_init.py [hunk 1]
```diff
@@ -60,6 +62,9 @@ def init_app(
         db.init_app(application)
     if login_manager:
         login_manager.init_app(application)
+        import dmutils.session
+        if os.environ.get('DM_USE_REDIS_SESSION_TYPE'):
```
Comment (lfdebrux) [inline dmutils/flask_init.py:66]: Could we just use the existing `SESSION_TYPE` config var here?

```suggestion
        if app.config.get("SESSION_TYPE") == "redis":
```

I think the advantage of this is 1) it doesn't need to stay as an environment variable later on 2) one less thing to remember 😁

**Comment by lfdebrux** [comment]
We should probably have one or two tests for this 🤔

File: dmutils/flask_init.py (hunk 1 above)
Comment (gidsg) [reply dmutils/flask_init.py:66]: @lfdebrux the reason I didn't use `SESSION_TYPE` here is that this allows us to do a gradual rollout of `SESSION_TYPE` to the frontend apps without breaking existing sessions across apps. 

Once this change is rolled out though, the condition could be changed as you suggest or potentially removed entirely.

Comment (lfdebrux) [reply dmutils/flask_init.py:66]: I see... why doesn't using `SESSION_TYPE` let us do a gradual rollout?

Comment (lfdebrux) [reply dmutils/flask_init.py:66]: Or rather, doesn't it amount to the same thing? If we can change `DM_USE_REDIS_SESSION_TYPE` in one go can't we change `SESSION_TYPE` in one go? Or am I missing something? 🤔

Comment (gidsg) [reply dmutils/flask_init.py:66]: @lfdebrux I found in testing that trying to set `SESSION_TYPE` globally e.g. using dmrunner settings required that it already exist with some value within the app config, else it was stripped out, so it was not set successfully. Using a `DM_` prefix env var is set globally though. Happy to talk this through more with you if that would help as I also found it confusing.

Comment (lfdebrux) [reply dmutils/flask_init.py:66]: Ah, I see. Well that makes sense then, let's go with this method you've already worked out!

**Comment by gidsg** [comment]
@lfdebrux I have added some tests now, please may you re-review? 
I am aware commit history is a bit messy but will rebase once approved.

**Review by lfdebrux — APPROVED** [review]

**Pull request merged**